File: ai_widget/languages/csharp/agent.cs
// C# / .NET Integration for Craveva AI Agent
// Install: dotnet add package Newtonsoft.Json

using System;
using System.Net.Http;
using System.Text;
using System.Threading.Tasks;
using Newtonsoft.Json;

public class CravevaAgentService
{
    private readonly string agentId = "69ccc35e7d0ece6ff702487b";
    private readonly string apiKey = "cvd_viPEIg-8UsVWKmO7hqfu4LZYxee4AoXE";
    private readonly string baseUrl = "https://ai.craveva.com";
    private readonly HttpClient httpClient;

    public CravevaAgentService(HttpClient httpClient)
    {
        this.httpClient = httpClient;
    }

    /// <summary>
    /// Send a message to the agent
    /// </summary>
    public async Task<AgentResponse> ChatAsync(string message, string userId = null, string companyId = null, string outletId = null)
    {
        var endpoint = $"{baseUrl}/api/v1/agents/{agentId}/chat";
        
        var payload = new
        {
            message = message,
            user_id = userId,
            company_id = companyId,
            outlet_id = outletId
        };

        var json = JsonConvert.SerializeObject(payload);
        var content = new StringContent(json, Encoding.UTF8, "application/json");

        var request = new HttpRequestMessage(HttpMethod.Post, endpoint)
        {
            Content = content
        };

        if (!string.IsNullOrEmpty(apiKey))
        {
            request.Headers.Authorization = new System.Net.Http.Headers.AuthenticationHeaderValue("Bearer", apiKey);
        }

        try
        {
            var response = await httpClient.SendAsync(request);
            response.EnsureSuccessStatusCode();
            
            var responseContent = await response.Content.ReadAsStringAsync();
            return JsonConvert.DeserializeObject<AgentResponse>(responseContent);
        }
        catch (Exception ex)
        {
            return new AgentResponse
            {
                Error = ex.Message,
                Output = "Sorry, I could not process that request."
            };
        }
    }
}

public class AgentResponse
{
    [JsonProperty("output")]
    public string Output { get; set; }
    
    [JsonProperty("error")]
    public string Error { get; set; }
}

// Usage example:
// var httpClient = new HttpClient();
// var agent = new CravevaAgentService(httpClient);
// var response = await agent.ChatAsync("Hello, how can you help me?");
// Console.WriteLine(response.Output);

File: ai_widget/languages/ruby/agent.rb
# Ruby Integration for Craveva AI Agent
# Install: gem install httparty

require 'httparty'
require 'json'

class CravevaAgent
  def initialize(agent_id = '69ccc35e7d0ece6ff702487b', api_key = 'cvd_viPEIg-8UsVWKmO7hqfu4LZYxee4AoXE', base_url = 'https://ai.craveva.com')
    @agent_id = agent_id
    @api_key = api_key
    @base_url = base_url
    @endpoint = "#{base_url}/api/v1/agents/#{agent_id}/chat"
  end

  def chat(message, user_id: nil, company_id: nil, outlet_id: nil)
    headers = {
      'Content-Type' => 'application/json'
    }
    
    headers['Authorization'] = "Bearer #{@api_key}" if @api_key

    payload = {
      message: message
    }
    
    payload[:user_id] = user_id if user_id
    payload[:company_id] = company_id if company_id
    payload[:outlet_id] = outlet_id if outlet_id

    begin
      response = HTTParty.post(@endpoint, {
        headers: headers,
        body: payload.to_json,
        timeout: 30
      })

      if response.success?
        JSON.parse(response.body)
      else
        { error: "HTTP #{response.code}: #{response.message}", output: 'Sorry, I could not process that request.' }
      end
    rescue => e
      { error: e.message, output: 'Sorry, I could not process that request.' }
    end
  end

  def get_response(message)
    result = chat(message)
    result['output'] || 'Sorry, I could not process that request.'
  end
end

# Usage example:
# agent = CravevaAgent.new(
#   '69ccc35e7d0ece6ff702487b',
#   'cvd_viPEIg-8UsVWKmO7hqfu4LZYxee4AoXE',
#   'https://ai.craveva.com'
# )
# response = agent.chat('Hello, how can you help me?')
# puts response['output']
